torch/csrc/jit/passes/xnnpack_rewrite.cpp

```diff
@@ -1,17 +1,83 @@
+#include <stack>
+
+#include <ATen/core/jit_type.h>
+#include <ATen/native/xnnpack/OpContext.h>
+
 #include <torch/csrc/jit/passes/constant_pooling.h>
 #include <torch/csrc/jit/passes/constant_propagation.h>
 #include <torch/csrc/jit/passes/xnnpack_rewrite.h>
 #include <torch/csrc/jit/passes/fuse_linear.h>
 #include <torch/csrc/jit/passes/graph_rewrite_helper.h>
 #include <torch/csrc/jit/passes/subgraph_rewrite.h>
 #include <torch/csrc/jit/ir/subgraph_matcher.h>
+#include <torch/csrc/jit/ir/ir.h>
 
 namespace torch {
 namespace jit {
 
 namespace {
 
 #ifdef USE_XNNPACK
+using at::native::xnnpack::XNNPackLinearOpContext;
+using at::native::xnnpack::XNNPackConv2dOpContext;
+
+bool nodeMatchesPackingOps(Node* n) {
+  return ((n->kind() == Symbol::fromQualString("xnnpack::linear_prepack")) ||
+      n->kind() == Symbol::fromQualString("xnnpack::conv2d_prepack"));
+}
+
+TypePtr getCustomType(Node* n) {
```

Comment:
We can get this type from the output of Node, right?

Reply:
You mean like, `node->outputs()[0]->type()`? Hmm, let me try.